From 5e8d38de38275201d23585669d892820d9f42636 Mon Sep 17 00:00:00 2001
From: arunkumarrajappa <arunkumarx.rajappa@intel.com>
Date: Mon, 5 Jun 2023 09:50:07 +0000
Subject: [PATCH] CTL: PC6_not_hitting_issue fix on system idle.

Issue :  PC6 is not hitting when System is in Idle state.
[Background -PC6 was working fine till ctl-17.0 but stopped working afterwards]

Fix: Enabled PCIe L1 support changes from 17.0 egs mainline branch to latest
catlow.

Tested: Tested with the changes on latest catlow mainline and validated the
scenario.
1. Set profile to powersave (using command tuned-adm profile powersave)
2. Run SOC tool on the host using command
   ./socwatch -m -f cpu-cstate -f cpu-pstate -t 300
3. Validate pc6 from generated output file of SOC Run.

Signed-off-by: arunkumarrajappa <arunkumarx.rajappa@intel.com>
Upstream-Status: Pending
---
 arch/arm/mach-aspeed/ast2600/platform.S | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm/mach-aspeed/ast2600/platform.S b/arch/arm/mach-aspeed/ast2600/platform.S
index 911b44ab9a..3ec9b07401 100644
--- a/arch/arm/mach-aspeed/ast2600/platform.S
+++ b/arch/arm/mach-aspeed/ast2600/platform.S
@@ -351,6 +351,20 @@ wait_lock:
 	movt    r1, #0x0000
 	str     r1, [r0]
 
+	/* enable PCIe L1 support */
+	ldr	r0, =0x1e6ed07c
+	ldr	r1, =0xa8
+	str	r1, [r0]
+	ldr	r0, =0x1e6ed010
+	ldr	r1, =0x27040fe1
+	str	r1, [r0]
+	ldr	r0, =0x1e6ed068
+	ldr	r1, =0xc81f0a
+	str	r1, [r0]
+	ldr	r0, =0x1e6ed07c
+	mov	r1, #0
+	str	r1, [r0]
+
 	/* MMIO decode setting */
 	ldr	r0, =AST_SCU_MMIO_DEC_SET
 	mov	r1, #0x2000
-- 
2.17.1

